java/daikon/dcomp/BuildJDK.java

```diff
@@ -90,288 +57,380 @@ protected boolean exists(String libRt) {
+        // Instrument the class file.
+        try {
+          build.processClassFile(jc, dest_dir, className);
+        } catch (Throwable e) {
+          throw new Error("Couldn't instrument " + className, e);
+        }
+      }
 
-      build.translate_classes(jfile, dest_dir, prefix, "");
+    } else { // translate from jar file or modules file
 
-      // Create the various helper classes
-      // build.dump_helper_classes(dest_dir);
+      if (BcelUtil.javaVersion > 8) {
+        build.translate_modules();
+      } else {
+        build.translate_jar();
+      }
+      build.translate_classes(dest_dir);
 
       // Write out the static map
-      System.out.printf("Found %d statics%n", DCInstrument.static_map.size());
+      System.out.printf("Found %d statics.%n", DCInstrument.static_map.size());
       DCInstrument.save_static_map(new File(dest_dir, static_map_fname));
 
       // Write out the list of all classes in the jar file
       File jdk_classes_dir = new File(dest_dir, "java/lang");
       File jdk_classes_file = new File(jdk_classes_dir, "jdk_classes.txt");
-      PrintWriter pw = new PrintWriter(jdk_classes_file, UTF_8.name());
       System.out.printf("Writing all classes to %s%n", jdk_classes_file);
-      for (String classname : all_classes) {
-        pw.println(classname);
+      try (PrintWriter pw = new PrintWriter(jdk_classes_file, UTF_8.name())) {
+        pw.println("no_primitives: " + DynComp.no_primitives);
+        for (String classname : all_classes) {
+          pw.println(classname);
+        }
       }
-      pw.flush();
-      pw.close();
 
       // Print out any methods that could not be instrumented
       print_skipped_methods();
     }
-    System.out.println("done at " + new Date());
+    System.out.println("BuildJDK done at " + new Date());
   }
 
   /**
-   * Check the resulting arguments for legality. Prints a message and returns false if there was an
-   * error.
+   * Check the arguments for legality. Verify java.home and JAVA_HOME match. Prints a message and
+   * returns false if there is an error.
    */
-  public static boolean check_args(Options options, String[] target_args) {
+  public static void check_args(Options options, String[] target_args) {
 
     if (classfiles) {
-      if (target_args.length < 2) {
-        System.out.println("must specify source jar and destination dir");
+      if (target_args.length < 1) {
+        System.out.println("must specify destination dir");
         options.printUsage();
-        return false;
+        System.exit(1);
       }
-      if (target_args.length < 3) {
+      if (target_args.length < 2) {
         System.out.println("must specify classfiles to instrument");
         options.printUsage();
-        return false;
+        System.exit(1);
       }
     } else {
-      if (target_args.length < 2) {
+      if (target_args.length < 1) {
         System.out.println("must specify source jar and destination dir");
```

Comment:
The two arguments seem to be in the wrong order:  doesn't the destination directory have to be first?

Reply:
error message was not updated.  should be "must specify destination dir"
fixed